nnvm/python/nnvm/frontend/pytorch/base.py

```diff
@@ -0,0 +1,119 @@
+r'''Basic classes for PyTorch operators and graphs'''
+from abc import ABC, abstractmethod
+from collections import OrderedDict
+from nnvm.symbol import Variable
+
+
+def _unique_name(node):
+    return next(node.outputs()).uniqueName()
+
+
+def _kind(node):
+    return node.kind()
+
+
+class PyTorchGraph:
+    r'''Wrapper for the PyTorch JIT IR graph'''
+
+    def __init__(self):
+        self.inputs = OrderedDict()
+        self.params = OrderedDict()
+        self.ops = OrderedDict()
+
+    def __getitem__(self, name):
+        if name in self.inputs:
+            return self.inputs[name]
+        elif name in self.params:
+            return self.params[name]
+        elif name in self.ops:
+            return self.ops[name]
+        else:
+            raise RuntimeError('Node {} not found.'.format(name))
+
+    def add_input(self, name, tensor):
+        self.inputs[name] = PyTorchInput(name, tensor)
+        self.inputs[name].graph = self
+
+    def add_param(self, name, tensor):
+        self.params[name] = PyTorchParam(name, tensor.astype('float32'))
+        self.params[name].graph = self
```

Comment:
`PyTorchParam` does not have property `graph`.

Reply:
Ditto